# Normalimenggunakan Min-Max Scaling dan Standardization (Z-Score Normalization) untuk beberapa fitur numerik

# Membaca dataset

In [1]:
import pandas as pd
from sklearn.preprocessing import MinMaxScaler, StandardScaler

# Load dataset
file_path = "D:\MATA KULIAH\SEMESTER 6\DATA SCIENCE\Week 6\InggridCantik\SuperMarket Analysis.csv"
df = pd.read_csv(file_path)

# Menampilkan beberapa baris pertama dataset
df.head()


<>:5: SyntaxWarning: invalid escape sequence '\M'
<>:5: SyntaxWarning: invalid escape sequence '\M'
C:\Users\User\AppData\Local\Temp\ipykernel_20864\1813379335.py:5: SyntaxWarning: invalid escape sequence '\M'
  file_path = "D:\MATA KULIAH\SEMESTER 6\DATA SCIENCE\Week 6\InggridCantik\SuperMarket Analysis.csv"


Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428   Alex     Yangon        Member  Female   
1  226-31-3081   Giza  Naypyitaw        Normal  Female   
2  631-41-3108   Alex     Yangon        Normal  Female   
3  123-19-1176   Alex     Yangon        Member  Female   
4  373-73-7910   Alex     Yangon        Member  Female   

             Product line  Unit price  Quantity   Tax 5%     Sales       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

          Time      Payment    cogs  gross margin percentage  gross income  \
0   1:08:00 PM      Ewallet  522.83                 4.761905       26.1415   
1  10:29:00 AM         Cash   76.40                 4.761905        3.8200   
2   1:23:00 PM  Credit card  324.31                 4.761905       16.2155   
3   8:33:00 PM      Ewallet  465.76                 4.761905       23.2880   
4  10:37:00 AM      Ewallet  604.17                 4.761905       30.2085   

   Rating  
0     9.1  
1     9.6  
2     7.4  
3     8.4  
4     5.3

In [3]:
# Memilih fitur numerik yang akan dinormalisasi
numeric_features = ["Unit price", "Quantity", "Tax 5%", "Sales", "cogs", "gross income", "Rating"]

# Min-Max Scaling (0-1)
min_max_scaler = MinMaxScaler()
df_minmax = df.copy()
df_minmax[numeric_features] = min_max_scaler.fit_transform(df[numeric_features])

# Standardization (Z-Score Normalization)
standard_scaler = StandardScaler()
df_standardized = df.copy()
df_standardized[numeric_features] = standard_scaler.fit_transform(df[numeric_features])

# Menampilkan hasil normalisasi pertama untuk dibandingkan
df_comparison = df[numeric_features].head(5).copy()
df_comparison["Unit price (MinMax)"] = df_minmax["Unit price"].head(5)
df_comparison["Unit price (Z-Score)"] = df_standardized["Unit price"].head(5)

df_comparison


Unit price  Quantity   Tax 5%     Sales    cogs  gross income  Rating  \
0       74.69         7  26.1415  548.9715  522.83       26.1415     9.1   
1       15.28         5   3.8200   80.2200   76.40        3.8200     9.6   
2       46.33         7  16.2155  340.5255  324.31       16.2155     7.4   
3       58.22         8  23.2880  489.0480  465.76       23.2880     8.4   
4       86.31         7  30.2085  634.3785  604.17       30.2085     5.3   

   Unit price (MinMax)  Unit price (Z-Score)  
0             0.718847              0.718160  
1             0.057855             -1.525303  
2             0.403316             -0.352781  
3             0.535603              0.096214  
4             0.848131              1.156959

Hasil Normalisasi
1. Min-Max Scaling
=>  Mengubah data ke skala 0 hingga 1.
    Contoh: Unit price awal 74.69 → menjadi 0.7188.

2. Z-Score Normalization (Standardization)
=>  Mengubah data sehingga memiliki mean = 0 dan standar deviasi = 1.
    Contoh: Unit price awal 74.69 → menjadi 0.7182.